ngraph/test/type_prop/gelu.cpp

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "gtest/gtest.h"
+#include "ngraph/ngraph.hpp"
+#include "util/type_prop.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+TEST(type_prop, gelu_default_mode_inference_f32)
+{
+    auto param = make_shared<op::Parameter>(element::f32, Shape{1, 32, 32});
+    auto gelu = make_shared<op::v7::Gelu>(param);
+
+    ASSERT_EQ(gelu->get_element_type(), element::f32);
+    ASSERT_EQ(gelu->get_shape(), (Shape{1, 32, 32}));
+    ASSERT_EQ(gelu->get_approximation_mode(), op::GeluApproximationMode::ERF);
+}
+
+TEST(type_prop, gelu_default_mode_inference_f16)
+{
+    auto param = make_shared<op::Parameter>(element::f16, Shape{1, 32, 32});
+    auto gelu = make_shared<op::v7::Gelu>(param);
+
+    ASSERT_EQ(gelu->get_element_type(), element::f16);
+    ASSERT_EQ(gelu->get_shape(), (Shape{1, 32, 32}));
+    ASSERT_EQ(gelu->get_approximation_mode(), op::GeluApproximationMode::ERF);
+}
+
+TEST(type_prop, gelu_tanh_mode_inference_f32)
+{
+    auto param = make_shared<op::Parameter>(element::f32, Shape{1, 32, 32});
+    auto gelu = make_shared<op::v7::Gelu>(param, op::GeluApproximationMode::TANH);
+
+    ASSERT_EQ(gelu->get_element_type(), element::f32);
+    ASSERT_EQ(gelu->get_shape(), (Shape{1, 32, 32}));
+    ASSERT_EQ(gelu->get_approximation_mode(), op::GeluApproximationMode::TANH);
+}
+
+TEST(type_prop, gelu_tanh_mode_inference_f16)
+{
+    auto param = make_shared<op::Parameter>(element::f16, Shape{1, 32, 32});
+    auto gelu = make_shared<op::v7::Gelu>(param, op::GeluApproximationMode::TANH);
+
+    ASSERT_EQ(gelu->get_element_type(), element::f16);
+    ASSERT_EQ(gelu->get_shape(), (Shape{1, 32, 32}));
+    ASSERT_EQ(gelu->get_approximation_mode(), op::GeluApproximationMode::TANH);
+}  
+
+TEST(type_prop, gelu_incompatible_input_type_boolean)
+{
+    auto param = make_shared<op::Parameter>(element::boolean, Shape{1, 32, 32});
+    ASSERT_THROW(std::make_shared<op::v7::Gelu>(param), ngraph::NodeValidationFailure);
+}
+
+TEST(type_prop, gelu_incompatible_input_type_u16)
```

Comment:
I see in spec that only floating point types are supported. If it is the case, then we should check that also other integer types are throwing.